### PR TITLE
wrapping fetch to send more informative errors to sentry

### PR DESCRIPTION
**Pull request by wheresrhys**
@matthew-andrews

### CODE REVIEW

File: src/js-setup.js [hunk 1]
```diff
@@ -17,6 +17,12 @@ JsSetup.noop = function () {};
 
 JsSetup.prototype.init = function (opts) {
 
+	// Wait til here to include as it has a dependency on Promise
+	// Won't break in its current incarnation, but should be careful anyway
+	require('isomorphic-fetch');
```
Comment (matthew-andrews) [inline src/js-setup.js]: don't think this change was necessary… but it's ok

File: src/js-setup.js [hunk 2]
```diff
@@ -26,10 +32,24 @@ JsSetup.prototype.init = function (opts) {
 	return flags.init({ url: opts.flagsUrl || '/__flags.json' }).then(function() {
 
 		if (flags.get('clientErrorReporting')) {
+			if (flags.get('clientAjaxErrorReporting')) {
+
+				var realFetch = window.fetch;
+
+				window.fetch = function (url, opts) {
+					return realFetch.call(this, url, opts)
```
Comment (matthew-andrews) [inline src/js-setup.js]: i guess you've already tested in chrome where it is native and it's ok?

Comment (wheresrhys) [inline src/js-setup.js]: haven't tested yet - just an idea so far, but good point. Will make sure

Comment (wheresrhys) [inline src/js-setup.js]: yep - is fine in canary

**Pull request merged**